Assets/Scripts/Mixed/Systems/KinematicCharacterControllerSystems.cs

```diff
@@ -14,59 +16,57 @@ namespace PropHunt.Mixed.Systems
     /// <summary>
     /// Updates the grounded data on a kinematic character controller
     /// </summary>
-    [BurstCompile]
     [UpdateInGroup(typeof(GhostPredictionSystemGroup))]
-    public class KCCGroundedSystem : ComponentSystem
+    public class KCCGroundedSystem : SystemBase
     {
         /// <summary>
         /// Maximum degrees between ground and player 
         /// </summary>
         public static readonly float MaxAngleFallDegrees = 90;
 
-        protected override unsafe void OnUpdate()
+        protected unsafe override void OnUpdate()
         {
-            BuildPhysicsWorld physicsWorld = World.GetExistingSystem<BuildPhysicsWorld>();
-            CollisionWorld collisionWorld = physicsWorld.PhysicsWorld.CollisionWorld;
+            PhysicsWorld physicsWorld = World.GetExistingSystem<BuildPhysicsWorld>().PhysicsWorld;
 
-            // Update grounded data for each KCC
-            Entities.ForEach((
-                Entity ent,
-                ref PhysicsCollider collider,
-                ref Translation trans,
-                ref Rotation rot,
+            Entities.WithBurst().ForEach((
```

Comment:
I don't think the explicit WithBurst is needed, but if it doesn't hurt then leave it there.
I couldn't find anything on the docs other than it's used to override the Burst compilation parameters.

Reply:
We'll know when we get the performance test working :) Thanks for the suggestion. It's a lot more readable and these didn't need the custom IJobChunk data.